### DISCUSSION TITLE
Background2D running on IRF with low statistics (few bins only) gives very different results between 1.0 and 0.17

### DISCUSSION THREAD

**Discussion by vkulikovskiy** [gammapy/gammapy#4345] [Help] [answered]
Behaviour:
v0.17 - nearest values are not 0 in "evaluate" with "nearest" 
v1.0 - nearest values are 0 in "evaluate" with "nearest" and few orders of magnitude lower than in 0.17 for "evaluate+nearest" and "integrate_log_log" vs "evaluate_integrate".

It looks something fishy may happen for the evaluation on the 2D grid, i.e. for 0.17 non 0 value is searched while 1.0 is just nearest. It could be that grids for some reason are not the same (although axis check does not reveal it).

It is not clear if such behaviour is known and it is correct in v1.0 (but then wrong in 0.17?).

Minimal example:

0.17 

```
bkg_mu = Background2D.read('../IRFs/bkg_mu.fits')
bg = bkg_mu.evaluate(
        fov_lon = 75*u.deg,  #deg
        fov_lat = 0*u.deg, # does not matter
        energy_reco = [1.,2]*u.TeV,
        method='nearest'
    )
print(bg)

bgint = bkg_mu.evaluate_integrate(
        fov_lon = 75*u.deg,  #deg
        fov_lat = 0*u.deg, # does not matter
        energy_reco = [1,2]*u.TeV,
        method = 'linear'   
    )
print(bgint)
```

Output:
```
[7.92064348e-13 4.03204528e-13] 1 / (MeV s sr)
[5.53973795e-13] TeV / (MeV s sr)

```

v1.0
```
bkg_mu = Background2D.read('../IRFs/bkg_mu.fits')
bg = bkg_mu.evaluate(
        offset = 75*u.deg,
        energy = [1.,2]*u.TeV,
        method='nearest'
    )
print(bg)




bgint = bkg_mu.integrate_log_log(
        'energy',
        offset = 75*u.deg,
        energy = [1,2]*u.TeV,
    )
print(bgint)
```

Output:
```
[0. 0.] 1 / (MeV s sr)
[9.95405601e-27] TeV / (MeV s sr)
```

Here is the fits file (the extension is .fits, I've added .txt just to upload it here).
[bkg_mu.fits.txt](https://github.com/gammapy/gammapy/files/10794532/bkg_mu.fits.txt)

**Comment by tunbehaun273** [comment, marked as answer]
The default `values_scale` of the bkg class in v1.0 is 'log'. This means that it interpolates on the log of the data array. If the data array includes many zeros this leads to problems. 
Inserting the line `bkg_mu.interp_kwargs['values_scale'] = 'lin'` after you create the bkg class instance should fix the issue.

**Comment by vkulikovskiy** [comment]
Thanks for support. In fact the value of integral becomes:
```
[5.53973777e-13] TeV / (MeV s sr)
```
which is much closer
